### PR TITLE
Deprecate repair-malformed-updates flag, move object meta mutation into BeforeCreate

### LINKED ISSUE #23297: Stop UID defaulting in ValidateObjectMetaUpdate
When handling PUT request, if the UID field is empty in the new object, API server defaults the UID of the new object to that of the existing object: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L344-L346

This was introduced to fix #14126 from the server side. A client-side fix is introduced in https://github.com/kubernetes/kubernetes/pull/15268.

I think users should always supply the UID, because otherwise the update might end up being applied on a different object that bears the same name, which could happen if an object is deleted and then recreated.

I think we should stop performing the defaulting in head, and inform the users to turning it on if they are using a pre-1.1 client.

@lavalamp @bgrant0607 @liggitt Do you have concerns?

### PR DESCRIPTION
**Pull request by liggitt**
closes #23297

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go [hunk 1]
```diff
@@ -254,39 +250,6 @@ func ValidateObjectMetaUpdate(newMeta, oldMeta *metav1.ObjectMeta, fldPath *fiel
...
-		}
-	}
-
-	// TODO: needs to check if newMeta==nil && oldMeta !=nil after the repair logic is removed.
-	if newMeta.GetDeletionGracePeriodSeconds() != nil && (oldMeta.GetDeletionGracePeriodSeconds() == nil || *newMeta.GetDeletionGracePeriodSeconds() != *oldMeta.GetDeletionGracePeriodSeconds()) {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("deletionGracePeriodSeconds"), newMeta.GetDeletionGracePeriodSeconds(), "field is immutable; may only be changed via deletion"))
-	}
-	if newMeta.GetDeletionTimestamp() != nil && (oldMeta.GetDeletionTimestamp() == nil || !newMeta.GetDeletionTimestamp().Equal(oldMeta.GetDeletionTimestamp())) {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("deletionTimestamp"), newMeta.GetDeletionTimestamp(), "field is immutable; may only be changed via deletion"))
-	}
```
Comment (tallclair) [inline staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go]: These last 2 checks aren't necessary anymore?

Comment (liggitt) [reply staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go]: the two fields are verified immutable below

**Comment by liggitt** [comment]
service registry was calling ValidateServiceUpdate for its side-effects... changed that to rest.BeforeCreate(..., service.Strategy). @smarterclayton PTAL

**Comment by liggitt** [comment]
@smarterclayton friendly ping

**Comment by liggitt** [comment]
@kubernetes/sig-api-machinery-pr-reviews

**Pull request merged**